src/store/reducers/AnimateMap.ts

```diff
@@ -0,0 +1,135 @@
+import {
+  AnimateMapActions,
+  AnimateMapActionTypes,
+} from "../actions/AnimateMap";
+import { Reducer } from "redux";
+import * as PIXI from "pixi.js";
+import { Box, QuadTree } from "js-quadtree";
+
+//Note: maybe relocate to src/types/AnimateMapVenue in the future
+// Gets the length of an array/tuple type.
+// see: https://dev.to/kjleitz/comment/gb5d
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type LengthOfTuple<T extends any[]> = T extends { length: infer L } ? L : never;
+
+// Drops the first element of a tuple.
+// see: https://dev.to/kjleitz/comment/gb5d
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type DropFirstInTuple<T extends any[]> = T extends [arg: any, ...rest: infer U]
+  ? U
+  : T;
+
+// Gets the type of the last element of a tuple.
+// see: https://dev.to/kjleitz/comment/gb5d
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type LastInTuple<T extends any[]> = T[LengthOfTuple<DropFirstInTuple<T>>];
+
+export type AnimateMapStageOptions = LastInTuple<
+  ConstructorParameters<typeof PIXI.Application>
+>;
+
+export type AnimateMapWorldBounds = { width: number; height: number };
+
+export interface AnimateMapPoint {
+  x: number; //integer
+  y: number; //integer
+}
+
+export interface ReplicatedUserData {
+  id: string;
+  videoUrlString: string;
+  avatarUrlString: string;
+  dotColor: number; //hex
+}
+
+export interface ReplicatedUser extends AnimateMapPoint {
+  data: ReplicatedUserData;
+}
+
+export interface ReplicatedVenueData {
+  id: string;
+  videoUrlString: string;
+  imageUrlString: string;
+}
+
+export interface ReplicatedVenue extends AnimateMapPoint {
+  data: ReplicatedVenueData;
+}
+
+interface AnimateMapState {
+  stageOptions: AnimateMapStageOptions;
+  worldWidth: number;
+  worldHeight: number;
+  zoom: number;
+  users: Map<string, ReplicatedUser>;
+  usersQT: QuadTree | null;
+  venues: Map<string, ReplicatedVenue>;
+  venuesQT: QuadTree | null;
+}
+
+const initialAnimateMapState: AnimateMapState = {
+  stageOptions: {
+    backgroundColor: 0x10bb99,
```

Comment:
Is there a reason we're using a hex number here instead of a hex string/similar?